### ISSUE TITLE
go struct output of list

### ISSUE THREAD

**Issue by hangxie** [hangxie/parquet-tools#216]
schema command has wrong go struct output on LIST types, eg:

raw schema:
```
    {
      "repetition_type": "OPTIONAL",
      "name": "CavBuList",
      "num_children": 1,
      "converted_type": "LIST",
      "logicalType": {
        "LIST": {}
      },
      "children": [
        {
          "repetition_type": "REPEATED",
          "name": "Array",
          "num_children": 12,
          "children": [
            {
              "type": "BYTE_ARRAY",
              "repetition_type": "OPTIONAL",
              "name": "CavBuId",
              "converted_type": "UTF8",
              "logicalType": {
                "STRING": {}
              }
            },
...
```

go struct:
```
	CavBuList        *[]*string `parquet:"name=CavBuList, type=LIST, convertedtype=LIST, repetitiontype=OPTIONAL"`
```

**Comment by hangxie** [comment]
This seems to be a bug introduced by https://github.com/hangxie/parquet-tools/pull/215, as old version is good:

```
	CavBuList        *[]*string `parquet:"name=CavBuList, type=LIST, valuetype=BYTE_ARRAY, valueconvertedtype=UTF8, convertedtype=LIST, repetitiontype=OPTIONAL"`
```

**Comment by hangxie** [comment]
I think the correct behavior is either to refuse to output go struct with complicated struct, or output:

```
	CavBuList        *[]*string `parquet:"name=CavBuList, type=LIST, valuetype=BYTE_ARRAY, valueconvertedtype=STRUCT, repetitiontype=OPTIONAL"`
```

FYI JSON output:
```
    {
      "Tag": "name=CavBuList, type=LIST, convertedtype=LIST, repetitiontype=OPTIONAL",
      "Fields": [
        {
          "Tag": "name=Element, type=STRUCT, repetitiontype=REQUIRED",
          "Fields": [
            {
              "Tag": "name=CavBuId, type=BYTE_ARRAY, convertedtype=UTF8, repetitiontype=OPTIONAL"
            },
            {
              "Tag": "name=Name, type=BYTE_ARRAY, convertedtype=UTF8, repetitiontype=OPTIONAL"
            },
            {
              "Tag": "name=AdminName, type=BYTE_ARRAY, convertedtype=UTF8, repetitiontype=OPTIONAL"
            },
            {
              "Tag": "name=AdminEmail, type=BYTE_ARRAY, convertedtype=UTF8, repetitiontype=OPTIONAL"
            },
            {
              "Tag": "name=CcoId, type=BYTE_ARRAY, convertedtype=UTF8, repetitiontype=OPTIONAL"
            },
...
```

**Comment by hangxie** [comment]
fixed by #217
